Sources/Filters/Core/Cutter/index.js

```diff
@@ -0,0 +1,272 @@
+          newPointID: -1, // id of the intersected point when it will be added into vtkPoints
+        });
+      }
+
+      // Add points into newPointList
+      for (let i = 0; i < intersectedEdgesList.length; i++) {
+        const intersectedEdge = intersectedEdgesList[i];
+        let alreadyAdded = false;
+        // Check if point/edge already added
+        for (let j = 0; j < crossedEdges.length; j++) {
+          const crossedEdge = crossedEdges[j];
+          const sameEdge =
+            intersectedEdge.pointEdge1 === crossedEdge.pointEdge1 &&
+            intersectedEdge.pointEdge2 === crossedEdge.pointEdge2;
+          const samePoint =
+            intersectedEdge.intersectedPoint[0] ===
+              crossedEdge.intersectedPoint[0] &&
+            intersectedEdge.intersectedPoint[1] ===
+              crossedEdge.intersectedPoint[1] &&
+            intersectedEdge.intersectedPoint[2] ===
+              crossedEdge.intersectedPoint[2];
+          if (sameEdge || samePoint) {
+            alreadyAdded = true;
+            intersectedEdgesList[i].newPointID = crossedEdges[j].newPointID;
+            break;
+          }
+        }
+        if (!alreadyAdded) {
+          newPointsData.push(intersectedEdge.intersectedPoint[0]);
+          newPointsData.push(intersectedEdge.intersectedPoint[1]);
+          newPointsData.push(intersectedEdge.intersectedPoint[2]);
+          intersectedEdgesList[i].newPointID = newPointsData.length / 3 - 1;
+          crossedEdges.push(intersectedEdgesList[i]);
+        }
+      }
+
+      // Store cells
+      const cellSize = intersectedEdgesList.length;
+      if (cellSize === 2) {
+        newLinesData.push(
+          cellSize,
+          intersectedEdgesList[0].newPointID,
+          intersectedEdgesList[1].newPointID
+        );
+      } else if (cellSize > 2) {
+        newPolysData.push(cellSize);
+        intersectedEdgesList.forEach((edge) => {
+          newPolysData.push(edge.newPointID);
+        });
+      }
+    }
+
+    // Set points
+    const outputPoints = output.getPoints();
+    outputPoints.setData(newPointsData);
+    if (outputPoints.getNumberOfComponents !== 3) {
+      outputPoints.setNumberOfComponents(3);
+    }
+
+    // Set lines
+    if (newLinesData.length !== 0) {
+      output.getLines().setData(newLinesData);
+    }
+
+    // Set polys
+    if (newPolysData.length !== 0) {
+      output.getPolys().setData(newPolysData);
+    }
+  }
+
+  // expose requestData
+  publicAPI.requestData = (inData, outData) => {
+    // implement requestData
+    const input = inData[0];
+
+    if (!input) {
+      vtkErrorMacro('Invalid or missing input');
+      return;
+    }
+
+    if (!model.cutFunction) {
+      vtkErrorMacro('Missing cut function');
+      return;
+    }
+
+    const output = vtkPolyData.newInstance();
+
+    dataSetCutter(input, output);
+
+    outData[0] = output;
+  };
+}
```

Comment:
I think you are missing the getMTime() that's why you call `modified()` on it when changing the implicit function.

Reply:
see https://github.com/Kitware/vtk-js/blob/master/Sources/Common/DataModel/ImplicitBoolean/index.js#L25-L35

Reply:
fixed